File: TPs JAVA et Angular/AngularEtSpring/src/main/java/fr/dta/AngularEtSpring/service/PonyService.java
package fr.dta.AngularEtSpring.service;

import java.util.ArrayList;
import java.util.List;

import fr.dta.AngularEtSpring.model.Pony;

public class PonyService {
	private List<Pony> ponies;

	public PonyService() { 
		this.ponies = new ArrayList<Pony>();
	}

	public List<Pony> getAllPonies()
	{
		return this.ponies;
	}

	public void addPony(Pony p)
	{
		this.ponies.add(p);
	}
}

File: TPs JAVA et Angular/littlepony/src/app/pony.service.ts
import { Injectable } from '@angular/core';
import { Pony } from './pony';
import { PONIES } from './ponies-mock';
import { Observable, of } from 'rxjs';
import { HttpClient, HttpHeaders } from '@angular/common/http';
import { Router } from '@angular/router';
import { RaceService } from './race.service';

@Injectable({
  providedIn: 'root'
})
export class PonyService
{ // equivalent a la DAO

  url: string = 'http://localhost:8085/api/ponies';
  httpOptions =
  {
    headers: new HttpHeaders({'Content-type': 'application/json'})
  };


  ponies: Array<Pony>;
  constructor(private http: HttpClient, private router: Router, private serviceRace: RaceService)
  { 
      this.ponies = [];
      this.ponies = PONIES;
  }


  getAllPonies(): Observable<Array<Pony>>
  {
      return  this.http.get<Array<Pony>>(this.url + '/', this.httpOptions);
  }

  addPony(p: Pony) : void
  {
    this.ponies.push(p);
  }
}
